language/move-analyzer/src/symbols.rs

```diff
@@ -2190,6 +2198,228 @@ pub fn on_hover_request(context: &Context, request: &Request, symbols: &Symbols)
     );
 }
 
+/**
+ * Returns a tuple where the first value is the index of the text info of the function name and the second value is the text info of the function
+ */
+fn find_closet_fn_in_text_info_vec<'a>(
+    start_idx: usize,
+    text_info_vec: &'a Vec<TextInfo>,
+) -> Option<(usize, TextInfo<'a>)> {
+    let mut parenthesis_delta = 0;
+
+    // Plus 1 to include the character at the starting index
+    for i in (0..start_idx + 1).rev() {
+        let text_info = text_info_vec.get(i).unwrap();
+
+        match text_info.text {
+            "(" => {
+                if parenthesis_delta == 0 {
+                    // The identifier to the left of the left parenthesis is the function.
+                    match text_info_vec.get(i - 1) {
+                        Some(text_info) => {
+                            return Some((i - 1, *text_info));
+                        }
+                        None => {
+                            return None;
+                        }
+                    }
+                } else {
+                    parenthesis_delta -= 1;
+                }
+            }
+            ")" => {
+                parenthesis_delta += 1;
+            }
+            _ => {}
+        }
+    }
+
+    None
+}
+
+/**
+ * Find the position of the active parameter given a vector of parsed tokens
+ */
+fn find_active_parameter<'a>(
+    // The starting index must be to 1 position to the right of a left parenthesis
+    start_idx: usize,
+    // end_idx >= start_idx
+    end_idx: usize,
+    text_info_vec: &'a Vec<TextInfo>,
+) -> u32 {
+    let mut parenthesis_delta = 0;
+    let mut active_parameter = 0;
+
+    for i in start_idx..end_idx {
+        let text_info = text_info_vec.get(i).unwrap();
+
+        match text_info.text {
+            "," => {
+                // When the parenthesis_delta equals to 0, it means we are in the correct function scope
+                if parenthesis_delta == 0 {
+                    active_parameter += 1;
+                }
+            }
+            "(" => {
+                parenthesis_delta += 1;
+            }
+            ")" => {
+                // This means we are at the end of the current function scope
+                if parenthesis_delta == 0 {
+                    break;
+                }
+
+                parenthesis_delta -= 1;
+            }
+            _ => {}
+        }
+    }
+
+    active_parameter
+}
+
+fn collect_parameter_info_from_ident(ident: &IdentType) -> Vec<ParameterInformation> {
```

Comment:
Please dd documentation for the function.

Reply:
done